lib/api/messages.js

```diff
@@ -1983,18 +1961,52 @@ module.exports = (db, server, messageHandler, userHandler, storageHandler, setti
     );
 
     server.post(
-        '/users/:user/mailboxes/:mailbox/messages/:message/forward',
+        {
+            path: '/users/:user/mailboxes/:mailbox/messages/:message/forward',
+            validationObjs: {
+                pathParams: {
+                    user: Joi.string().hex().lowercase().length(24).required(),
+                    mailbox: Joi.string().hex().lowercase().length(24).required(),
+                    message: Joi.number().required()
+                },
+                queryParams: {},
+                requestBody: {
+                    target: Joi.number().min(1).max(1000),
+                    addresses: Joi.array().items(Joi.string().email({ tlds: false })),
+                    sess: sessSchema,
+                    ip: sessIPSchema
+                },
+                response: {
+                    200: {
+                        description: 'Success',
+                        model: Joi.object({
+                            success: Joi.boolean(),
+                            queueId: Joi.string(),
+                            forwarded: Joi.array().items(
+                                Joi.object({
+                                    seq: Joi.string(),
+                                    type: Joi.string(),
+                                    value: Joi.string()
+                                }).$_setFlag('objectName', 'Forwarded')
+                            )
+                        })
```

Comment:
There are a couple things I am still thinking of adding but highly likely not with this given PR:
1. Joi has checks for hex, binary as well as string max and min length and all of this seems to be supported by OpenAPI as well so adding that would add even more clarity to the inputs (as well as outputs)
2. Noticed that some string inputs in the current doc are actually string `Enum`s, so I am planning to add that as well, currently that functionality is missing and even if string is an enum it will be displayed as generic string type in the docs

Support for example strings would be nice indeed, but currently am not really aware of whether Joi has that kind of thing built in, perhaps will just have to add it do description and hope that the renderer renders it correctly. Will research!

Other responses besides status 200 are to be supported later on as well actually, but they require more work (or maybe not :D, not sure yet about how to best structure them)

Reply:
> Additionally, being able to specify a custom openapi schema would be nice. It wouldn't be used normally, but there's some edgecases where you need the openapi spec to reflect something that's just not possible to define with Joi.

Could be added, sure

Reply:
Yeah, enums are the way you can define "it's one of these simple values" in an openapi spec. It's very helpful for docs and typed languages!

I did find this:
https://joi.dev/api/?v=17.9.1#anyexampleexample-options

The doc tool we use will use these openapi examples on the right side, instead of something like `email: 'string'` it'd automatically fill it with `email: 'john@example.com'`. It works quite nicely!